webapp/src/js/controllers/contacts.js

```diff
@@ -209,52 +209,69 @@ var _ = require('underscore'),
       } else {
         title = ContactSchema.get(selected.doc.type).label;
       }
+      $scope.loadingSummary = true;
       return $q
         .all([
           $translate(title),
           getActionBarDataForChild(selectedDoc.type),
           getCanEdit(selectedDoc),
-          ContactSummary(selected.doc, selected.reports, selected.lineage),
-          Settings()
         ])
         .then(function(results) {
           $scope.setTitle(results[0]);
           if (results[1]) {
             selectedDoc.child = results[1];
           }
           var canEdit = results[2];
-          var summary = results[3];
-          $scope.selected.summary = summary;
-          var options = { doc: selectedDoc, contactSummary: summary.context };
-          XmlForms('ContactsCtrl', options, function(err, forms) {
-            if (err) {
-              $log.error('Error fetching relevant forms', err);
-            }
-            var showUnmuteModal = function(formId) {
-              return $scope.selected.doc.muted && !isUnmuteForm(results[4], formId);
-            };
-            var formSummaries =
-              forms &&
-              forms.map(function(xForm) {
-                return {
-                  code: xForm.internalId,
-                  title: translateTitle(xForm.translation_key, xForm.title),
-                  icon: xForm.icon,
-                  showUnmuteModal: showUnmuteModal(xForm.internalId)
+
+          $scope.setRightActionBar({
+            relevantForms: [], // REVIEWER: this disables the "New Action" button in action bar until full load is complete
+            selected: [selectedDoc],
+            sendTo: selectedDoc.type === 'person' ? selectedDoc : '',
+            canDelete: false, // REVIEWER: this disables the "Delete" button in action bar until full load is complete
+            canEdit: canEdit,
+          });
+
+          return selected.reportLoader.then(function() {
+            return $q.all([
+              ContactSummary(selected.doc, selected.reports, selected.lineage),
+              Settings()
+            ])
+            .then(function(results) {
```

Comment:
Consider improving readibility by replacing `results` here with `[summary, settings]` in the argument list for the function.